Confusing error message when trying to run a query statement with Exec

When executing a `SELECT` statement with `db.Exec` the error returned the error returned is a bit confusing, it claims that it is not an error.
```Go
_, err = db.Exec("SELECT name FROM person")
if err != nil {
	panic(err)
}
```
Gives
```
panic: not an error
```
The full error object appears to be:
```
protocol.Error{Code:100, Message:"not an error"}
```

--- a/src/gateway.c
+++ b/src/gateway.c
@@ -429,6 +429,9 @@ static const char *error_message(sqlite3 *db, int rc)
 			return "disk I/O error";
 		case SQLITE_ABORT:
 			return "abort";
+		case SQLITE_ROW:
+			return "rows yielded when none expected for EXEC "
+			       "request";
 	}
 
 	return sqlite3_errmsg(db);

--- a/test/unit/test_gateway.c
+++ b/test/unit/test_gateway.c
@@ -1180,6 +1180,26 @@ TEST_CASE(exec, manyParams, NULL)
 	return MUNIT_OK;
 }
 
+TEST_CASE(exec, unexpectedRow, NULL)
+{
+	struct exec_fixture *f = data;
+	uint64_t stmt_id;
+	(void)params;
+	CLUSTER_ELECT(0);
+	EXEC("CREATE TABLE test (n INT)");
+	EXEC("INSERT INTO test (n) VALUES (1)");
+	PREPARE("SELECT n FROM test");
+	f->request.db_id = 0;
+	f->request.stmt_id = stmt_id;
+	ENCODE(&f->request, exec);
+	HANDLE(EXEC);
+	WAIT;
+	ASSERT_CALLBACK(0, FAILURE);
+	ASSERT_FAILURE(SQLITE_ROW,
+		       "rows yielded when none expected for EXEC request");
+	return MUNIT_OK;
+}
+
 /******************************************************************************
  *
  * query

